### ISSUE TITLE
grunt task fails with "Unable to find local grunt" (moodle-local_chatlogs specific)

### ISSUE THREAD

**Issue by stronk7** [moodlehq/moodle-plugin-ci#45]
For some reason, moodle-plugin-ci (v3) is having problems with "moodle-local_chatlogs", failing with grunt not available error when running the grunt task). See my "test" branch failing:

https://travis-ci.com/github/stronk7/moodle-local_chatlogs/builds/190839966

I've triple checked [moodle-local_chatlogs' .travis.yml](https://github.com/stronk7/moodle-local_chatlogs/blob/e0e9fd1f3f4fac4863144b37e7d23abc15136564/.travis.yml), comparing with other plugins that are passing and also with the docs and I've run out of ideas. Everything looks ok there.

Also, i've executed moodle-ci-install with `-vvv` to get the installation details and see if there is something strange there:

https://travis-ci.com/github/stronk7/moodle-local_chatlogs/builds/190841587

^^^ Apparently, grunt is installed ok, more yet `grunt ignorefiles` is executed ok, but, later, `mooldle-plugin-ci` fails with "Unable to find local grunt" message.

Need a couple of eyes, sure there is something obvious, but I'm failing to find it, grrr!

Ciao :-)

**Comment by andrewnicols** [comment]
There are a couple of things going on here.

Firstly, the package required to install the global grunt executable is `grunt-cli`, but moodle-plugin-ci is installing `grunt` (which is the local part that a particular package should use). As it happens, `moodle-plugin-ci` is using the binary in the `node_modules` directory anyway (which is not correct), but that isn't the bug here. Weirdly, when I do fix that we get an error thrown about the symlink already existing. I'm not sure where that's come from so I'm not going to touch that particular brokenness in this issue.

This message "Unable to find local grunt" comes because the working directory does not contain a valid Gruntfile.js, and nor do any of its parent directories.

Looking at the structure of this repository I think that the actual reason behind this is somewhat stupid.

In ye olden days, YUI was located in `path/to/plugin/yui/[filename].js`, but about 8 years ago we started running Shifter on our JS to minify it and compile them as proper modules. When we did this we moved things to `path/to/plugin/yui/src/modname/[filename].js`.

The moodle-local_chatlogs plugin is configured with only the old version, but the code in moodle-plugin-ci does this:

```
            case 'yui':
                $yuiDir = $this->plugin->directory.'/yui';
                if (!is_dir($yuiDir)) {
                    return null;
                }

                return new GruntTaskModel($task, $yuiDir.'/src', 'yui/build');
```

Essentially it's assuming that if the `yui` directory exists, it must run in `yui/src`.

In this case there is no `yui/src` directory, and when Grunt is passed a non-existent directory as a workingDirectory, it does not traverse up to the parent directory (it can't... it's in a non-existent working directory and doesn't know the path) so we get this error.

Adding some debugging to moodle-plugin-ci we see this more clearly.

```
I am a Grunt 'yui' task with a WD of /home/travis/build/moodle and a build dir of 
Checking directory for task 'yui'
YUI Task found
YUI Directory found
I am a Grunt 'yui' task with a WD of /home/travis/build/moodle/local/chatlogs/yui/src and a build dir of yui/build
grunt-cli: The grunt command line interface (v1.3.2)
Fatal error: Unable to find local grunt.
If you're seeing this message, grunt hasn't been installed locally to
your project. For more information about installing and configuring grunt,
please see the Getting Started guide:
https://gruntjs.com/getting-started
```

The task was `yui`
The YUI directory was found (`/home/travis/build/moodle/local/chatlogs/yui`)
The GruntTaskModel was created with a directory of `/home/travis/build/moodle/local/chatlogs/yui/src`

That directory doesn't exist.

I think that the best solution here is to change the task to look for `yui/src` rather than `yui` in the plugin directory. There is no point running Grunt on older-style directories anyway because Grunt doesn't actually perform any checks for them anyway - it only works on a src directory.

So the solution is to modify the check to look for a `yui/src` directory.
